### DISCUSSION TITLE
Identical plot CARMA 10 and 21

### DISCUSSION THREAD

**Discussion by karandogra987** [ywx649999311/EzTao#96] [Announcements]
I am trying to fit CARMA model to my data. Using log-likelihood values, I found that the CARMA (2,1) process is the best. However, when plotting the CARMA(2,1) and CARMA(1,0) using the plot_pred_lc module, both the models produce exactly the same fitted LC. I have checked for whole light curves and individual points that both models generate the same fit, so I wanted to know (if possible) what could be the reasons for different log-likelihood values then?
Log-likelihood values :
CARMA(1,0) ~ 888
CARMA(2,1) ~ 920

PS: Unlike in the plot tutorial, I didn't use t_pred for the points that didn't exist; I just gave t_pred as the original timestamps.
I am attaching a screenshot of the fits. The fits are plotted using blue and red colors but because they are overlapping it produces a somewhat violet color.
![Screenshot from 2024-12-31 14-22-21](https://github.com/user-attachments/assets/f852114c-415d-4556-8c8b-8df3338b06c2)

**Comment by ywx649999311** [comment]
Hi, 

I don't think the different log-likelihood values is a big concern. When you do model comparison, you typically use AIC or BIC to penalize more complex models. But I want to make sure that there is no hidden bug in the package. Can you check the following:

1. What are your best-fit CARMA(2,1) and CARMA(1,0) parameters?
2. Can you use `eztao.ts.carma_sim.pred_lc` to verify if the predicated values at your data points are **EXACTLY** the same for the two models?

**Comment by karandogra987** [comment]
Hi,

You are correct about using AIC or BIC values, which I used to select the best model. The only concern I had was the same plot. As per the checks you mentioned above:

1) parameters for CARMA(1,0) : [0.0223, 0.1412]
    parameters for CARMA(2,1) : [0.2781, 0.0022, 0.0221, 0.1499]

2) I have checked the predicted values for both models. The values are not exactly the same. The distribution of the difference in predicted values peaks at the change in values after four decimal places.

This may resolve the issue, as I looked at the plots explicitly, which will definitely miss the variation in the predicted values mentioned above.

**Comment by ywx649999311** [comment]
OK, I think it makes more sense that the predicated values are fairly close but not exactly the same. You could also create model PSDs following this [tutorial](https://eztao.readthedocs.io/en/latest/notebooks/05_OtherUtils.html). If the PSDs of these two models are very similar, then it makes even more sense that the predicated values are so close. Nonetheless, if both models give reasonable fits, then their predicated values shouldn't be too different at your input data points.

**Comment by ywx649999311** [comment]
I am converting this thread to a discussion, as it doesn't seem to be any bug related to the code?
